sdk/resourcemanager/Azure.ResourceManager.Core/README.md

```diff
@@ -14,46 +14,159 @@ Install-Package Azure.ResourceManager.Core -Version 1.0.0-beta.1
+
+In order to reduce both the number of clients needed to perform common tasks and the amount of redundant parameters that each of those clients take, we have introduced an object hierarchy in the SDK that mimics the object hierarchy in Azure. Each resource client in the SDK has methods to access the resource clients of its children that is already scoped to the proper subscription and resource group.
+
+To accomplish this we are introducing 4 standard types for all resources in Azure.
+#### **[Resource]Data**
+This represents the data that makes up a given resource.  Typically this is the response data from a service call such as GET and provides details about the underlying resource.
+Previously this was represented by a **Model** class.
+
+#### **[Resource]Operations**
+
+This represents a service client that is scoped to a particular resource.
+You can directly execute all operations on that client without needing to pass in scope
+parameters such as subscription id or resource name.
+
+#### **[Resource]Container**
+
+This represents the operations you can perform on a collection of resources belonging to a specific parent resource.
+This mainly consists of List or Create operations.
+For most things the parent will be a **ResourceGroup**, however each parent / child relationship is represented this way.
+Such as a **Subnet** is a child of a **VirtualNetwork** or a **ResourceGroup** is a child of a **Subscription**.
+
+#### **[Resource]**
 
-Key concepts of the Azure .NET SDK can be found [here](https://azure.github.io/azure-sdk/dotnet_introduction.html)
+This represents a full resource object which contains a **Data** property exposing the details as a **[Resource]Data** type.
+It also has access to all of the operations and like the **[Resource]Operations** object is already scoped
+to a specific resource in Azure.
 
-## Documentation
+### Migrating from current to preview versions
+There are core differences between the current and new preview SDK.
+The main ones are:
+- The introduction of Resource Operations.
+- An structured Resource Identifier.
 
-Documentation is available to help you learn how to use this package
+You can learn more about the main differences [in this document](docs\MigratingFromCurrentToPreview.md) and reading the examples below.
 
-- [Quickstart](https://github.com/Azure/azure-sdk-for-net/blob/master/doc/mgmt_preview_quickstart.md)
-- [API References](https://docs.microsoft.com/dotnet/api/?view=azure-dotnet)
-- [Authentication](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/README.md)
 
 ## Examples
+### Add a tag to a virtual machine
+ Imagine that our company requires all virtual machines to be tagged with the owner and we are tasked with writing a program to add the tag to any missing virtual machines in a given resource group.
+
+ ```csharp
+// First we construct our armClient
+ArmClient armClient = new ArmClient(new DefaultAzureCredential());
```

Comment:
Unless someone feels strongly about it, I vote for only using the explicit type in code samples when the type isn't clear. Thoughts?

Reply:
@m-nash what are your thoughts on this?

I don't have a preference about this, but my first impression is that putting explicit types could help to understand how the hierarchy works and know exactly the type that each method returns. Although, it is noticeably clear in most places.

Reply:
@scottaddie which type is not explicit in the line that is linked here?

Reply:
I was advocating for the use of `var` when instantiating the `ArmClient`.